File: DSA/Pragya_Agarwal/Day16_Question1.py
class Node:
    def __init__(self,val):
        self.data = val
        self.left = None
        self.right = None

def postOrder(node):
    if node == None:
        return 
    
    postOrder(node.left)
    postOrder(node.right)
    print(node.data,end=" ")

if __name__ == '__main__':
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.left.left = Node(4)
    root.left.right = Node(5)

print("Postorder Traversal:")
postOrder(root)
    

File: DSA/Manav-Rajput/Day16_question1.java
import java.util.ArrayList;
import java.util.List;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

public class BinaryTreePostorderTraversal {

    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        postorderHelper(root, result);
        return result;
    }

    private void postorderHelper(TreeNode node, List<Integer> result) {
        if (node == null) {
            return;
        }
        postorderHelper(node.left, result);
        postorderHelper(node.right, result);
        result.add(node.val);
    }

    public static void main(String[] args) {
        // Create the binary tree from the example
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);
        root.left.right = new TreeNode(5);

        BinaryTreePostorderTraversal solution = new BinaryTreePostorderTraversal();
        List<Integer> postorder = solution.postorderTraversal(root);
        System.out.println(postorder);  // Output: [4, 5, 2, 3, 1]
    }
}
